src/hotspot/share/cds/cdsConfig.cpp

```diff
@@ -429,8 +429,9 @@ bool CDSConfig::check_vm_args_consistency(bool patch_mod_javabase, bool mode_fla
   }
 
   if (is_dumping_static_archive()) {
-    if (!mode_flag_cmd_line) {
-      // By default, -Xshare:dump runs in interpreter-only mode, which is required for deterministic archive.
+    if (!mode_flag_cmd_line JVMCI_ONLY(&& !UseJVMCICompiler && !UseGraalJIT && !EnableJVMCIProduct && !EnableJVMCI)) {
```

Comment:
I would suggest expanding the comment to explain why so many JVMCI flags need to be checked here.

Reply:
I added the following comments

```
// We check the JVMCI flags here to avoid running with -Xint when the JVMCI compiler is used.
// Ideally we should only check EnableJVMCI, but that flag can be ergonomically enabled
// in CompilerConfig::check_args_consistency(), which is called after the current function returns.
// So we have to manually check all the relevant flags here. TODO: This should be refactored.
```